### ISSUE TITLE
sometimes a query window can't be closed

### ISSUE THREAD

**Issue by aisbergde** [microsoft/azuredatastudio#14168]
Issue Type: <b>Bug</b>

I can't close some Query Windows. I use Ctrl-4 or the context menu of the window, I get the dialog (Save, Don't Save, Cancel). I press "Don't Save", but the window doen't close. This happens only in one specific window. It happens all the time, but I am afraid, this can't be reproduced, because I see this the first time)

Azure Data Studio version: azuredatastudio 1.25.2 (4d4917d328ab7a7526817a0f0ef03237a79aa54d, 2021-01-20T22:52:13.492Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz (4 x 2904)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.87GB (8.88GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (24)</summary>

Extension|Author (truncated)|Version
---|---|---
deletedb|Ale|0.0.3
tsqlchecker|Dan|0.0.1
qpi|joc|0.1.6
newdatabase|kev|1.0.0
admin-pack|Mic|0.0.2
admin-tool-ext-win|Mic|0.0.2
agent|Mic|0.48.0
cms|Mic|0.8.0
dacpac|Mic|1.6.0
import|Mic|1.2.0
machine-learning|Mic|0.6.0
powershell|ms-|2020.4.0
profiler|Mic|0.11.0
query-history|Mic|0.1.0
schema-compare|Mic|1.8.0
server-report|Mic|0.2.3
sql-assessment|Mic|0.5.0
sql-database-projects|Mic|0.5.1
schema-visualization|R0t|0.3.0
sql-prompt|Red|0.2.6
sql-search|Red|0.4.0
simple-data-scripter|sea|0.1.3
vscodeintellicode|Vis|1.2.1
poor-sql-formatter|WSR|0.1.0

</details>

**Comment by aisbergde** [comment]
I closed and reopened ADS and then it was closed.

**Comment by alanrenmsft** [comment]
@aisbergde did you happen to use the compare feature? if so, this could happen, I will fix the issue of because of file compare.

**Comment by aisbergde** [comment]
@alanrenmsft yes, sometimes I use the compare feature, comparing current and previous git versions.
